tests/Unit/Domain/Creators/Test_RefinedCylinder.cpp

```diff
@@ -0,0 +1,410 @@
+#include "Parallel/RegisterDerivedClassesWithCharm.hpp"
+#include "Utilities/MakeArray.hpp"
+
+namespace domain {
+namespace {
+
+void test_refined_cylinder_boundaries(const bool use_equiangular_map) {
+  INFO("Refined Cylinder boundaries");
+  // definition of an arbitrary refined cylinder
+  const double inner_radius = 2.0;
+  const double outer_radius = 6.0;
+  const double lower_bound = -2.5;
+  const double upper_bound = 5.0;
+  const std::vector<double> radial_partitioning = {4.0};
+  const std::vector<double> height_partitioning = {1.25};
+  const size_t refinement_level = 2;
+  const std::array<size_t, 3> expected_wedge_extents{{5, 4, 3}};
+  const bool is_periodic_in_z{false};
+  const std::vector<std::array<size_t, 3>> expected_refinement_level{
+      (1 + 4 * (1 + radial_partitioning.size())) *
+          (1 + height_partitioning.size()),
+      make_array<3>(refinement_level)};
+  const creators::Cylinder refined_cylinder{inner_radius,
+                                            outer_radius,
+                                            lower_bound,
+                                            upper_bound,
+                                            is_periodic_in_z,
+                                            refinement_level,
+                                            expected_wedge_extents,
+                                            use_equiangular_map,
+                                            radial_partitioning,
+                                            height_partitioning};
+  test_physical_separation(refined_cylinder.create_domain().blocks());
+
+  const auto domain = refined_cylinder.create_domain();
+  const OrientationMap<3> aligned_orientation{};
+  const OrientationMap<3> quarter_turn_ccw(std::array<Direction<3>, 3>{
+      {Direction<3>::lower_eta(), Direction<3>::upper_xi(),
+       Direction<3>::upper_zeta()}});
+  const OrientationMap<3> half_turn(std::array<Direction<3>, 3>{
+      {Direction<3>::lower_xi(), Direction<3>::lower_eta(),
+       Direction<3>::upper_zeta()}});
+  const OrientationMap<3> quarter_turn_cw(std::array<Direction<3>, 3>{
+      {Direction<3>::upper_eta(), Direction<3>::lower_xi(),
+       Direction<3>::upper_zeta()}});
+  std::vector<DirectionMap<3, BlockNeighbor<3>>> expected_block_neighbors{};
+  std::vector<std::unordered_set<Direction<3>>> expected_external_boundaries{};
+  using TargetFrame = Frame::Inertial;
+  std::vector<std::unique_ptr<
+      domain::CoordinateMapBase<Frame::Logical, TargetFrame, 3>>>
+      coord_maps{};
+  expected_block_neighbors = std::vector<DirectionMap<3, BlockNeighbor<3>>>{
+      // Block 0 - layer 0 center 0
+      {{Direction<3>::upper_xi(), {1, aligned_orientation}},
+       {Direction<3>::upper_eta(), {2, quarter_turn_ccw}},
+       {Direction<3>::lower_xi(), {3, half_turn}},
+       {Direction<3>::lower_eta(), {4, quarter_turn_cw}},
+       {Direction<3>::upper_zeta(), {0 + 9, aligned_orientation}}},
```

Comment:
A comment reminding the reader where these `+9`s, etc come from would be good to have